### PR TITLE
Fixed most of the Portability Report issues

### PR DESCRIPTION
**Pull request by mwhelan**
Took dependency on Json.Net which is hopefully only temporary.
Outstanding issues are AppDomain unload and StackTrace.

### CODE REVIEW

File: src/TestStack.BDDfy/Processors/UnusedExampleException.cs [hunk 1]
```diff
@@ -1,21 +1,12 @@
 ﻿namespace TestStack.BDDfy.Processors
 {
     using System;
-    using System.Runtime.Serialization;
 
-    [Serializable]
```
Comment (JakeGinnivan) [inline src/TestStack.BDDfy/Processors/UnusedExampleException.cs]: You could polyfill this if you wanted. Removing for everyone means you can no longer take exception across appdomains

Comment (mwhelan) [inline src/TestStack.BDDfy/Processors/UnusedExampleException.cs]: By polyfill, do you mean #if statements?

Comment (JakeGinnivan) [inline src/TestStack.BDDfy/Processors/UnusedExampleException.cs]: Create an attribute in your code in the right namespace. Does nothing but don't need ifs.

Comment (mwhelan) [inline src/TestStack.BDDfy/Processors/UnusedExampleException.cs]: OK, I see what you mean. But then I would still need an #if for the serializable constructor?

**Pull request merged**